posthog/auth.py

```diff
@@ -68,8 +68,11 @@ def authenticate(cls, request: Union[HttpRequest, Request]) -> Optional[Tuple[An
             )
         except PersonalAPIKey.DoesNotExist:
             raise AuthenticationFailed(detail=f"Personal API key found in request {source} is invalid.")
-        personal_api_key_object.last_used_at = timezone.now()
-        personal_api_key_object.save()
+
+        now = timezone.now()
+        if personal_api_key_object.last_used_at is None or now.hour != personal_api_key_object.last_used_at.hour:
```

Comment:
Oh LOL thanks for catching this. For context, this is just so that the at-most-hourly update frequency is kept, while providing users with fully accurate information (even if less precise than before).